test-support/src/main/java/org/testcontainers/testsupport/FlakyTestJUnit4RetryRule.java

```diff
@@ -0,0 +1,72 @@
+package org.testcontainers.testsupport;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.MultipleFailureException;
+import org.junit.runners.model.Statement;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * TODO: Javadocs
+ */
+@Slf4j
+public class FlakyTestJUnit4RetryRule implements TestRule {
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        final Flaky annotation = description.getAnnotation(Flaky.class);
+        if (annotation != null) {
+            if ( ! isReviewDatePassed(annotation)) {
```

Comment:
Funny how this code would IMO be easier to read if the method is not extracted:
```java
if (LocalDate.now().isBefore(reviewDate)) {
   // retry
}
```

Reply:
Fair point - removes an unnecessary inversion